src/screens/auth/VerifyEmail/VerifyEmail.tsx

```diff
@@ -4,26 +4,50 @@ import { View } from 'react-native';
 import { Text } from 'src/components';
 import { ScreenTemplate } from 'src/components/templates';
 import { RootStackParamList } from 'src/navigation';
+import { useAppDispatch } from 'src/store';
+import { AsyncThunks } from 'src/store/thunks';
 
 import styles from './VerifyEmail.style';
 
-const VerifyEmail = () => {
+const VerifyEmail = ({ route }: any) => {
   const navigation = useNavigation<NavigationProp<RootStackParamList>>();
+  const dispatch = useAppDispatch();
+  const [optionalMessage, setOptionalMessage] = React.useState('Waiting...');
+
+  const verification = async () => {
+    const response = await dispatch(AsyncThunks.signUp(route.params));
+    if (response.payload?.status === 200) {
+      setOptionalMessage('Link sent to your email adress \nYou can sign in after verification');
+      setTimeout(() => {
+        navigation.navigate('Signin');
+      }, 5000);
+    }
+    if (response.payload?.status === 400) {
+      setOptionalMessage('Email already exists');
+      setTimeout(() => {
+        navigation.navigate('Main');
```

Comment:
Avoid using string literals. Value might be changed in a while and you will need to replace it everywhere.

I understand that in that specific case, TS does the types check to avoid unacceptable values, but it would be better to create constants.